geostructures/structures.py

```diff
@@ -959,6 +961,11 @@ def __init__(
         self.nw_bound = nw_bound
         self.se_bound = se_bound
 
+        if not isinstance(nw_bound, Coordinate):
```

Comment:
Is there a functional way to abstract some of this? Like if you had a function along the lines of:
```
_ensure_types([nw_bound, se_bound], [Coordinate, Coordinate])
```
It would probably reduce the amount of code needed to check each individual type